baseplate/secrets/fetcher.py

```diff
@@ -282,13 +282,19 @@ def is_about_to_expire(self):
     def get_secret(self, secret_name):
         """Get the value and expiration time of a named secret."""
         logger.debug("Fetching secret %r.", secret_name)
-        response = self.session.get(
-            urljoin(self.base_url, posixpath.join("v1", secret_name)),
-            headers={
-                "X-Vault-Token": self.token,
-            },
-        )
-        response.raise_for_status()
+        try:
+            response = self.session.get(
+                urljoin(self.base_url, posixpath.join("v1", secret_name)),
+                headers={
+                    "X-Vault-Token": self.token,
+                },
+            )
+            response.raise_for_status()
+        except requests.HTTPError:
```

Comment:
I do have a question here. In the documentation it looks like the exception would be `requests.exceptions.HTTPError`, but on this [page](https://2.python-requests.org/en/master/api/#requests.Response.raise_for_status)) it says it calls the HTTPError, and links to `https://2.python-requests.org/en/master/api/#requests.HTTPError`. So I just want to make sure this is the correct syntax, as there is no easy way to test.

Reply:
The exception is defined in `requests.exceptions` so that's its canonical name, but it is also [imported into `requests`](https://github.com/kennethreitz/requests/blob/master/requests/__init__.py#L120) so available as `requests.HTTPError` as well. It's the same class object either way so it doesn't matter which name you use.

You should be able to test this in the OneVM by writing a short config file for a secret fetcher that points at the test Vault available there. See the integration tests here in baseplate for example usage.